src/test/unit/StrategyManagerUnit.t.sol

```diff
@@ -271,34 +458,212 @@ contract StrategyManagerUnitTests is Test, Utils {
+        cheats.expectRevert();
+        strategyManager.depositIntoStrategy(strategy, token, amount);
+        cheats.stopPrank();
+    }
+
+    function test_Revert_WhenTokenDoesNotExist() external {
+        // replace 'dummyStrat' with one that uses a non-existent token
+        dummyToken = IERC20(address(5678));
+        dummyStrat = _deployNewStrategy(dummyToken, strategyManager, pauserRegistry, dummyAdmin);
+
+        // whitelist the strategy for deposit
+        cheats.startPrank(strategyManager.owner());
+        IStrategy[] memory _strategy = new IStrategy[](1);
+        _strategy[0] = dummyStrat;
+        cheats.expectEmit(true, true, true, true, address(strategyManager));
+        emit StrategyAddedToDepositWhitelist(dummyStrat);
+        strategyManager.addStrategiesToDepositWhitelist(_strategy);
+        cheats.stopPrank();
+
+        address staker = address(this);
+        IERC20 token = dummyToken;
+        uint256 amount = 1e18;
+        IStrategy strategy = dummyStrat;
+
+        cheats.startPrank(staker);
+        cheats.expectRevert();
+        strategyManager.depositIntoStrategy(strategy, token, amount);
+        cheats.stopPrank();
+    }
+
+    function test_Revert_WhenStrategyDepositFunctionReverts() external {
+        // replace 'dummyStrat' with one that always reverts
+        dummyStrat = StrategyBase(address(new Reverter()));
+
+        // whitelist the strategy for deposit
+        cheats.startPrank(strategyManager.owner());
+        IStrategy[] memory _strategy = new IStrategy[](1);
+        _strategy[0] = dummyStrat;
+        cheats.expectEmit(true, true, true, true, address(strategyManager));
+        emit StrategyAddedToDepositWhitelist(dummyStrat);
+        strategyManager.addStrategiesToDepositWhitelist(_strategy);
+        cheats.stopPrank();
+
+        address staker = address(this);
+        IERC20 token = dummyToken;
+        uint256 amount = 1e18;
+        IStrategy strategy = dummyStrat;
+
+        cheats.startPrank(staker);
+        cheats.expectRevert();
+        strategyManager.depositIntoStrategy(strategy, token, amount);
+        cheats.stopPrank();
+    }
+
+    function test_Revert_WhenStrategyDoesNotExist() external {
+        // replace 'dummyStrat' with one that does not exist
+        dummyStrat = StrategyBase(address(5678));
+
+        // whitelist the strategy for deposit
+        cheats.startPrank(strategyManager.owner());
+        IStrategy[] memory _strategy = new IStrategy[](1);
+        _strategy[0] = dummyStrat;
+        cheats.expectEmit(true, true, true, true, address(strategyManager));
+        emit StrategyAddedToDepositWhitelist(dummyStrat);
+        strategyManager.addStrategiesToDepositWhitelist(_strategy);
+        cheats.stopPrank();
+
+        address staker = address(this);
+        IERC20 token = dummyToken;
+        uint256 amount = 1e18;
+        IStrategy strategy = dummyStrat;
+
+        cheats.startPrank(staker);
+        cheats.expectRevert();
```

Comment:
explicit revert reason

Reply:
For reverts that give the error `Reason: Call reverted as expected, but without data` how do you write the `expectRevert()`? This is the case for the two test cases I didn't update

Reply:
agreed, sometimes you have a revert without data and this is just the required syntax